gprofiler/utils/__init__.py

```diff
@@ -233,12 +233,14 @@ def run_process(
     reraise_exc: Optional[BaseException] = None
     with start_process(cmd, via_staticx, **kwargs) as process:
         try:
-            communicate_kwargs = dict(input=stdin) if stdin is not None else {}
+            if stdin is not None:
+                assert process.stdin is not None
+                process.stdin.write(stdin)
             if stop_event is None:
                 assert timeout is None, f"expected no timeout, got {timeout!r}"
                 if communicate:
                     # wait for stderr & stdout to be closed
-                    stdout, stderr = process.communicate(timeout=timeout, **communicate_kwargs)
+                    stdout, stderr = process.communicate()
```

Comment:
Not respecting the `timeout` parameter
```suggestion
                    stdout, stderr = process.communicate(timeout=timeout)
```

Reply:
asserted `None` before